### PR TITLE
feat: implement password hasher & hash the password for `User` class

### PR DESCRIPTION
**Pull request by catouberos**
## Describe your changes
- Implemented password hashing using PBKDF2
- Implemented hashing password for creating new password on class `User`
- Implemented password check on class `User`
- Introduced tests for password hashing

### CODE REVIEW

**Review by HippityHarry — APPROVED** [review]
I'm nothing but a puny human compared to this titan

File: src/main/java/pls_no_shinobu/videostore/model/User.java [hunk 1]
```diff
@@ -144,14 +146,18 @@ public void setUsername(String username) throws IllegalArgumentException {
         this.username = username;
     }
 
-    protected String getPassword() {
```
Comment (BuiAnhMinh) [inline src/main/java/pls_no_shinobu/videostore/model/User.java:147]: don't know why but this is red on my GitHub, and line 148 and line 154 too. But I think they are fine .-. Who am I to judge god himself.

**Review by BuiAnhMinh — APPROVED** [review]
just learn something new about red lines so no problem. Thank god for me being friend with this titan

**Review by HippityHarry — COMMENTED** [review]
the password hasher's lore deepened

**Pull request merged**